pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authorization/AuthorizationProvider.java

```diff
@@ -129,11 +142,15 @@ CompletableFuture<Boolean> canConsumeAsync(TopicName topicName, String role,
      *
      * @param topicName
      * @param role
-     * @return
-     * @throws Exception
+     *
+     * @deprecated
+     * Use {@link #allowTopicOperationAsync(TopicName, String, TopicOperation, AuthenticationDataSource)} instead.
      */
-    CompletableFuture<Boolean> canLookupAsync(TopicName topicName, String role,
-            AuthenticationDataSource authenticationData);
+    @Deprecated
+    default CompletableFuture<Boolean> canLookupAsync(TopicName topicName, String role,
+            AuthenticationDataSource authenticationData) {
+        return CompletableFuture.completedFuture(false);
+    }
```

Comment:
Instead of providing a default implementation, I think we should just deprecate these methods. Then, in the `PulsarAuthorizationProvider` class, we can remove the `@Override` annotation and those methods become internal implementation details instead of top level methods on the interface.